api/v1alpha1/clientprofile_types.go

```diff
@@ -31,10 +31,15 @@ type CephFsConfigSpec struct {
 
 	//+kubebuilder:validation:Optional
 	FuseMountOptions map[string]string `json:"fuseMountOptions,omitempty"`
+
+	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="radosNamespace is immutable"
+	//+kubebuilder:validation:Optional
+	RadosNamespace string `json:"radosNamespace,omitempty"`
 }
 
 // RbdConfigSpec defines the desired RBD configuration
 type RbdConfigSpec struct {
+	//+kubebuilder:validation:XValidation:rule="self == oldSelf",message="radosNamespace is immutable"
```

Comment:
Lets not do that. We might want to use changing this name when we unfortunately need to migrate data between rados namespaces